aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,94 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void axis_check(IntArrayRef& axis, int64_t bound) {
```

Comment:
just take a look how does other functions handle this axis condition spend roughly ~ 10 mins try to find out if there is anything else.

Reply:
there is a lambda function definition inside <<movedim>> function at TensorShape.cpp file.  (all_unique and wrap_dims). But they are not separate function.

Reply:
OK; just add a comment explaining what this function does and its intended use